shapely/tests/test_predicates.py

```diff
@@ -106,6 +111,53 @@ def test_binary_missing(func):
     assert (~actual).all()
 
 
+@pytest.mark.parametrize("a", all_types)
+@pytest.mark.parametrize("func, func_bin", XY_PREDICATES)
+def test_xy_array(a, func, func_bin):
+    with ignore_invalid(shapely.is_empty(a)):
+        # Empty geometries give 'invalid value encountered' in all predicates
+        # (see https://github.com/libgeos/geos/issues/515)
+        actual = func([a, a], (2, 3))
+        expected = func_bin([a, a], Point(2, 3))
+    assert actual.shape == (2,)
+    assert actual.dtype == np.bool_
+    np.testing.assert_allclose(actual, expected)
+
+
+@pytest.mark.parametrize("a", all_types)
+@pytest.mark.parametrize("func, func_bin", XY_PREDICATES)
+def test_xy_array_broadcast(a, func, func_bin):
+    with ignore_invalid(shapely.is_empty(a)):
+        # Empty geometries give 'invalid value encountered' in all predicates
+        # (see https://github.com/libgeos/geos/issues/515)
+        actual = func(a, np.array([[0, 1, 2], [1, 2, 3]]).T)
```

Comment:
This might be easier to visually compare to the expected result by using the transposed form directly:

```Python
actual = func(a, [[0,1], [1, 2], [2, 3]])
```

Reply:
Yes, sorry, that's mostly from quickly adapting the previous tests that were using separate x and y arrays to work with the latest commit to combine it in a single function (which then always needs interleaved). But if we go with separate functions, I will just revert that commit. 
And I am assuming that if we provide separate functions, we can support both separate x and y as interleaved, being consistent with the `shapely.points(..)` constructor.